In [1]:
#Jaccard coefficient

def jaccard_similarity_coefficient(list_a,list_b):

    set_intersection = set.intersection(set(list_a), set(list_b))

    num_intersection = len(set_intersection)


    set_union = set.union(set(list_a), set(list_b))

    num_union = len(set_union)

    #calculate Jaccard coefficient

    try:
        return float(num_intersection) / num_union
    except ZeroDivisionError:
        return 1.0

In [5]:
# make Lists

list_Jack = ["G_M", "F_Y", "C_N", "1_P", "2_N", "3_N", "4_A"]
list_Mary = ["G_F", "F_Y", "C_N", "1_P", "2_A", "3_P", "4_N"]
list_Jim = ["G_M", "F_Y", "C_P", "1_N", "2_N", "3_N", "4_A"]

In [6]:
# Pair1 (Jack, Mary)

if __name__== '__main__':
  Pair1 = jaccard_similarity_coefficient(list_Jack,list_Mary)
  print(Pair1)

0.2727272727272727


In [7]:
# Pair2 (Jack, Jim)

if __name__== '__main__':
  Pair2 = jaccard_similarity_coefficient(list_Jack,list_Jim)
  print(Pair2)

0.5555555555555556


In [8]:
# Pair3 (Jim, Mary)

if __name__== '__main__':
  Pair3 = jaccard_similarity_coefficient(list_Jim,list_Mary)
  print(Pair3)

0.07692307692307693


Pair1 (Jack, Mary)

Jaccard coefficient = 0.2727

Pair2 (Jack, Jim)

Jaccard coefficient = 0.5556

Pair3 (Jim, Mary)

Jaccard coefficient = 0.0769